src/Command/UnregisterAdherentsCommand.php

```diff
@@ -0,0 +1,164 @@
+
+            return 1;
+        }
+
+        $this->em->beginTransaction();
+
+        $this->unregisterAdherents($emails, $output);
+
+        $this->em->commit();
+
+        $output->writeln(PHP_EOL.'Unregistration finished.');
+
+        if ($this->notFoundEmails) {
+            $output->writeln('The following email addresses are not found in DB :');
+            foreach ($this->notFoundEmails as $email) {
+                $output->writeln($email);
+            }
+        }
+        if ($this->hosts) {
+            $output->writeln('The adherents with following email address are hosts of committees. They are not unregistered :');
+            foreach ($this->hosts as $email) {
+                $output->writeln($email);
+            }
+        }
+        if ($this->citizenProjectAdministrators) {
+            $output->writeln('The adherents with following email address are administrators of citizen projects. They are not unregistered :');
+            foreach ($this->citizenProjectAdministrators as $email) {
+                $output->writeln($email);
+            }
+        }
+    }
+
+    private function parseCSV(string $filename): array
+    {
+        if (false === ($handle = fopen($filename, 'r'))) {
+            throw new FileNotFoundException(sprintf('% not found', $filename));
+        }
+
+        $isFirstRow = true;
+        $emails = [];
+        while (false !== ($data = fgetcsv($handle, 100000, ';'))) {
+            if (true === $isFirstRow) {
+                $isFirstRow = false;
+
+                continue;
+            }
+
+            $row = array_map('trim', $data);
+            $emails[] = $row[4];
+        }
+        fclose($handle);
+
+        return $emails;
+    }
+
+    private function unregisterAdherents(array $emails, OutputInterface $output): void
+    {
+        $batchSize = 20;
+        $i = 1;
+        $unregistrationFactory = new UnregistrationFactory();
+        $progress = new ProgressBar($output, count($emails));
+        $progress->start();
+
+        foreach ($emails as $email) {
+            try {
+                if (!$adherent = $this->adherentRepository->findOneByEmail($email)) {
+                    $this->notFoundEmails[] = $email;
+
+                    continue;
+                }
+
+                if ($adherent->isHost()) {
+                    $this->hosts[] = $email;
+
+                    continue;
+                }
+
+                if ($adherent->isAdministrator()) {
+                    $this->citizenProjectAdministrators[] = $email;
+
+                    continue;
+                }
+
+                $unregistration = $unregistrationFactory->createFromAdherent($adherent);
+                $this->adherentRegistry->unregister($adherent, $unregistration);
+            } catch (\Exception $ex) {
+                throw new \RuntimeException(PHP_EOL.sprintf('An error occured while unregistering an adherent with email "%s": %s', $email, $ex->getMessage()));
```

Comment:
Please add `$ex` to the constructor as the `$previous` exception

Reply:
Done